src/map/utils/blueutils.cpp

```diff
@@ -438,57 +445,70 @@ namespace blueutils
                 {
                     CBlueTrait* PTrait = (CBlueTrait*)i;
 
-                    if (PTrait && PTrait->getCategory() == category)
+                    // Player is eligible for this Blue Trait
+                    if (PTrait && PTrait->getCategory() == category && totalWeight >= PTrait->getPoints())
                     {
                         bool add = true;
 
+                        // Check if any existing player Traits conflict
                         for (std::size_t j = 0; j < PChar->TraitList.size(); ++j)
                         {
                             CTrait* PExistingTrait = PChar->TraitList.at(j);
 
                             if (PExistingTrait->getID() == PTrait->getID())
                             {
-                                if (PExistingTrait->getLevel() == 0 && ((CBlueTrait*)PExistingTrait)->getCategory() == PTrait->getCategory())
+                                // Player has the real job trait, making them ineligible
+                                if (PExistingTrait->getLevel() > 0)
```

Comment:
hmm. BLU is capable of getting traits above the strength of main job traits, and easily able to get above the strength of subs.Iis this true of job mastered BLU? I can't say I know if it's true or not, however.

It was my understanding people set AoE BLU spells on retail that give them more MAB than /RDM does, so this could be pretty easily testable.

Reply:
Will approve anyway because this is a good improvement, and a bit beyond the scope of this PR

Reply:
Maybe it only overwrites if the blue trait is stronger? News to me that Blu traits would apply if the sj gave the trait

Reply:
https://www.bg-wiki.com/ffxi/Blue_Mage_Job_Traits -> `In the event that a trait is granted through both spells and one's support job, whichever grants the higher level of the trait will take priority.`

Reply:
Updated some comments and added a TODO